### ISSUE TITLE
NUnit Test Discoverer 3 in MS VisStudio 2015 gets a null object reference error when discovering

### ISSUE THREAD

**Issue by john-sc** [nunit/nunit3-vs-adapter#769]
Using MS Visual Studio Professional 2015
Visual c# 2015
Nunit version 3.12.0
Nunit3 version 3.15.1

I open a solution and it works. Tests are discovered.
I make any change to a line of code and rebuild.
I get
An exception occurred while test discoverer 'NUnit3TestDiscoverer' was loading tests. Exception: Object reference not set to an instance of an object.

**Comment by mikkelbu** [comment]
@john-sc I'm transferring this issue to the nunit3-vs-adapter repo (the original repo is for the NUnit 2 adaper).

**Comment by OsirisTerje** [comment]
@john-sc  What happens if you upgrade to 3.17 ?
If it is still there, can you upload a small repro?

**Comment by john-sc** [comment]
Hi,  thank you. I found my problem. I had version 3.15 installed using nuget package manager. And 3.17 installed using tools, extensions. I disabled the 3.17 and that fixed it. My error.   Regards, John

**Comment by prithvivinayak1** [comment]
I vae same problem. but not solved

**Comment by justcode411** [comment]
I have the same issue, it is extremely frustrating.
NUnit 3.12
NUnitAdapter 3.17
The only way to get it working once the error happens is to restart the compiler. And for large enterprise level apps, it can take 3-4 minutes to reload the solution and you need to do this every time you recompile a small code change if you want to test it.

**Comment by OsirisTerje** [comment]
@prithvivinayak1   @justcode411  John-sc solved https://github.com/nunit/nunit3-vs-adapter/issues/769#issuecomment-659122178 his problem by ensuring he didn't have any VSIX based adapters installed at the same time.   Can you check the same?

If not, please upload a small repro.  

Also, please set Verbosity to 5 (in a runsettings file), and enable the [DumpXmlTestDiscovery](https://docs.nunit.org/articles/vs-test-adapter/Tips-And-Tricks.html#dumpxmltestdiscovery-and-dumpxmltestresults).

**Comment by prithvivinayak1** [comment]
Using MS Visual Studio Professional 2015
Visual c# 2015
Nunit version 3.12.0
Nunit3 version 3.16
.Net framework 4.5.2
I opened a solution and it works. Tests are discovered.
I made some changes to a line of code and rebuild.
I got
An exception occurred while test discoverer 'NUnit3TestDiscoverer' was loading tests. Exception: Object reference not set to an instance of an object.
If i change default processor architecture(Test->Test settings ) from x86 to x64 or x64 to x86 it will work once. Same error occurs during the next rebuild
<img width="911" alt="Annotation 2020-07-20 151745" src="https://user-images.githubusercontent.com/68429564/88015508-fbbb7a00-cb3e-11ea-95b8-4d6f8f01068c.png">

**Comment by prithvivinayak1** [comment]
Using MS Visual Studio Professional 2015
Visual c# 2015
Nunit version 3.12.0
Nunit3 version 3.16
.Net framework 4.5.2
I opened a solution and it works. Tests are discovered.
I made some changes to a line of code and rebuild.
I got
An exception occurred while test discoverer 'NUnit3TestDiscoverer' was loading tests. Exception: Object reference not set to an instance of an object.
If i change default processor architecture(Test->Test settings ) from x86 to x64 or x64 to x86 it will work once. Same error occurs during the next rebuild
<img width="911" alt="Annotation 2020-07-20 151745" src="https://user-images.githubusercontent.com/68429564/88015627-45a46000-cb3f-11ea-8c47-647e746b96e3.png">

**Comment by prithvivinayak1** [comment]
> @prithvivinayak1 @justcode411 John-sc solved [#769 (comment)](https://github.com/nunit/nunit3-vs-adapter/issues/769#issuecomment-659122178) his problem by ensuring he didn't have any VSIX based adapters installed at the same time. Can you check the same?
> 
> If not, please upload a small repro.
> 
> Also, please set Verbosity to 5 (in a runsettings file), and enable the [DumpXmlTestDiscovery](https://docs.nunit.org/articles/vs-test-adapter/Tips-And-Tricks.html#dumpxmltestdiscovery-and-dumpxmltestresults).



> please set Verbosity to 5 (in a runsettings file), and enable the [DumpXmlTestDiscovery](https://docs.nunit.org/articles/vs-test-adapter/Tips-And-Tricks.html#dumpxmltestdiscovery-and-dumpxmltestresults).



> @prithvivinayak1 @justcode411 John-sc solved [#769 (comment)](https://github.com/nunit/nunit3-vs-adapter/issues/769#issuecomment-659122178) his problem by ensuring he didn't have any VSIX based adapters installed at the same time. Can you check the same?
> 
> If not, please upload a small repro.
> 
> Also, please set Verbosity to 5 (in a runsettings file), and enable the [DumpXmlTestDiscovery](https://docs.nunit.org/articles/vs-test-adapter/Tips-And-Tricks.html#dumpxmltestdiscovery-and-dumpxmltestresults).

@OsirisTerje  i set verbosity to 5 and enable DumpXmlTestDiscovery. Dump file is not created when i got the error

**Comment by prithvivinayak1** [comment]
Hi Thank you. I found my issue. 3.16 and 3.17 was installed.i disabled  Nunit3TestAdapter in Tools->Extensions and Updates.
Now its working

**Comment by OsirisTerje** [comment]
@justcode411  I would assume you have the same problem, nice if you can confirm.

**Comment by svetvasilev** [comment]
Hi, I experience the same issue even though it is marked as closed. My setup is VS 2015 Upd3, project targeting .NET 4.6. Test project was created with the NUnit VS templates add-on. Using nUnit v3.12, nUnit test adapter v3.17 through the nuget package only.
Initially I experienced the exact same behavior as above since I had the VSIX variant of the adapter installed as well. Tried to first disable it - no change. Removed it completely - the behaviour changed slightly. Now I am able to run tests more often than only once. yet after few runs the same error with the NullReferenceException upon test discovery occurs. Just closing the solution and opening again in the same VS instance seems to do the trick, but it is very unproductive.

**Comment by OsirisTerje** [comment]
Have you tried deleting the VSTest vsix cache ?   See https://github.com/nunit/nunit3-vs-adapter/issues/261#issuecomment-675926304

**Comment by svetvasilev** [comment]
@OsirisTerje - tried this one with some mixed outcome. Cleaned the cache several times as per #261 but the behaviour prevailed. Then I noticed that for some strange reason the NUnit adapter VS extension was still hanging around, though disabled. I uninstalled it then for good (for second time), restarted VS and then the adapter seemed to be loaded correctly and tests are possible to execute. 
However since then what is happening is that each time  when i try to run tests for very first time from context menu in code->Run Tests i still get the exact same exception. Same happens if i open the Test Explorer and try to run the tests from there. 
This is somewhat intermittent as once i restart VS the adapter starts behaving as expected. But i have to occasionally restart VS which is a bit annoying. 
Perhaps it has smthg to do with the tooling i have in this VS instance. I still use 2015 since i work with BizTalk server, which requires this VS version for the dev experience to function properly. I have as well SSDT and SSDT BI installed. I have in addition VS 2017 installed side-by-side.
